### DISCUSSION TITLE
Temperature graphs

### DISCUSSION THREAD

**Discussion by zachliu** [jgyates/genmon#1196] [Ideas]
I think temperature graphs make sense especially for people who's using the external sensors to monitor oil and battery temperatures. But after a quick check on the source code, this looks like a big task:
* wherever we have [PowerGraph](https://github.com/search?q=repo%3Ajgyates%2Fgenmon%20powergraph&type=code) defined, we'll need a new "TempGraph"
* wherever we have [PowerLog](https://github.com/search?q=repo%3Ajgyates%2Fgenmon+powerlog&type=code), we need a new "TempLog"

There are definitely other details. I'll start collecting them in this post.

**Comment by jgyates** [comment]
Hi @zachliu 

There are quite a few folks who are using grafana with influxdb and the MQTT genmon add on to create their own graphs like this:

<img width="1695" alt="Screenshot 2024-10-10 at 3 54 21 PM" src="https://github.com/user-attachments/assets/8a196d62-5cd4-4b7d-b76d-69866c1e086e">

Since I have gotten a few different requests to graph separate things I have opted to just default to MQTT and let folks roll their own rather than adding new graphs.

**Reply by zachliu** [reply]
mqtt is "ok" but ideally i'd want everything to be in one place
i think i'll start poking around and create the feature in my fork when i'm ready :saluting_face:

**Reply by zachliu** [reply]
@jgyates i finally did it! thanks to the almighty ai! :robot: 

https://github.com/zachliu/genmon/pull/1
<img width="1934" height="959" alt="genmon" src="https://github.com/user-attachments/assets/7128d161-6c82-4d19-8a5f-a6fb9ddc0929" />

**Reply by jgyates** [reply]
@zachliu 

Very nice. 

What happens if you enabled "Disable Platform Stats" on the advanced settings page and disable the temperature add on? This tests the situation where there is no temp to report. I am asking because this would be a factor in merging this into the genmon tree.

**Reply by jgyates** [reply]
fyi, No temp data is the default for non raspberry pi systems.

**Reply by zachliu** [reply]
@jgyates It degrades gracefully. With platform stats disabled and no temp addon, `GetTempSensorNames()` returns an empty list, the front-end creates no temperature chart tiles, no temp data is logged, and the existing power chart / other features are unaffected. The UI looks exactly as it does in 2.0.0 without this PR.

**Reply by jgyates** [reply]
@zachliu 

Thanks for taking the time to incorporate and test this. I have merged the PR from your branched tree back to the main genmon tree. My testing works as well. I also updated the backup and restore to include the nested folder that contains the temp logs (/etc/genmon/templogs). Let me know if you have any issues.
